File: src/main/java/com/example/climbingapi/dto/WallResponse.java
package com.example.climbingapi.dto;

import com.example.climbingapi.model.Wall;

import java.math.BigDecimal;
import java.time.OffsetDateTime;

public record WallResponse(
        Integer id,
        Integer areaId,
        String name,
        String description,
        BigDecimal latitude,
        BigDecimal longitude,
        String approachInfo,
        OffsetDateTime createdAt
) {

    public static WallResponse from(Wall wall) {
        return new WallResponse(
                wall.getId(),
                wall.getAreaId(),
                wall.getName(),
                wall.getDescription(),
                wall.getLatitude(),
                wall.getLongitude(),
                wall.getApproachInfo(),
                wall.getCreatedAt()
        );
    }
}

File: src/main/kotlin/com/example/climbingapi/dto/WallResponse.kt
package com.example.climbingapi.dto

import java.math.BigDecimal
import java.time.OffsetDateTime

data class WallResponse(
    val id: Int?,
    val areaId: Int?,
    val name: String?,
    val description: String?,
    val latitude: BigDecimal?,
    val longitude: BigDecimal?,
    val approachInfo: String?,
    val createdAt: OffsetDateTime?
)
